pkg/kyverno/test/command.go

```diff
@@ -0,0 +1,536 @@
+	sanitizederror "github.com/kyverno/kyverno/pkg/kyverno/sanitizedError"
+	v1 "github.com/kyverno/kyverno/pkg/api/kyverno/v1"
+	"github.com/kyverno/kyverno/pkg/kyverno/common"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	policy2 "github.com/kyverno/kyverno/pkg/policy"
+	"github.com/kyverno/kyverno/pkg/engine"
+	"github.com/kyverno/kyverno/pkg/engine/context"
+	"github.com/kyverno/kyverno/pkg/openapi"
+	"github.com/kyverno/kyverno/pkg/engine/response"
+	yamlv2 "gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"github.com/kyverno/kyverno/pkg/policyreport"
+	report "github.com/kyverno/kyverno/pkg/api/policyreport/v1alpha1"
+	"github.com/kyverno/kyverno/pkg/engine/utils"
+	ut "github.com/kyverno/kyverno/pkg/utils"
+	"github.com/kataras/tablewriter"
+	"github.com/lensesio/tableprinter"
+	"github.com/go-git/go-billy/v5/memfs"
+	"github.com/go-git/go-billy/v5"
+	"github.com/fatih/color"
+)
+
+// Command returns version command
+func Command() *cobra.Command {
+
+	var  valuesFile string
+	return &cobra.Command{
+		Use:   "test",
+		Short: "Shows current test of kyverno",
+		RunE: func(cmd *cobra.Command, dirPath []string) (err error) {
+			defer func() {
+				if err != nil {
+					if !sanitizederror.IsErrorSanitized(err) {
+						log.Log.Error(err, "failed to sanitize")
+						err = fmt.Errorf("internal error")
+					}
+				}
+			}()
+			err = testCommandHelper(dirPath, valuesFile)
+			if err != nil {
+				log.Log.V(3).Info("Test command fail to apply")
+				return err
+			}
+			return nil
+		},
+	}
+}
+
+type Test struct {
+	Name          string     `json:"name"`
+	Policies      []string     `json:"policies"`
+	Resources     []string     `json:"resources"`
+	Variables     string     `json:"variables"`
+	TResults      []TestResults     `json:"results"`
+}
+
+type SkippedPolicy struct {
+	Name     string    `json:"name"`
+	Rules    []v1.Rule `json:"rules"`
+	Variable string    `json:"variable"`
+}
+
+type TestResults struct {
+	Policy    string    `json:"policy"`
+	Rule  string `json:"rule"`
+	Status string    `json:"status"`
+	Resource string    `json:"resource"`	
+}
+
+type ReportResult struct {
+	TestResults 
+	Resources []*corev1.ObjectReference `json:"resources"`
+}
+
+type Resource struct {
+	Name          string     `json:"name"`
+	Values map[string]string `json:"values"`
+}
+
+type Table struct {
+	ID    int    `header:"#"`
+	Resource string    `header:"test"`
+	Result string    `header:"result"`
+	
+}
+type Policy struct {
+	Name      string     `json:"name"`
+	Resources []Resource `json:"resources"`
+}
+
+type Values struct {
+	Policies []Policy `json:"policies"`
+}
+
+func testCommandHelper(dirPath []string, valuesFile string) (err error) {
+	var errors []error
+	fs := memfs.New()
+	
+	if len(dirPath) == 0 {
+			return  sanitizederror.NewWithError(fmt.Sprintf("require test yamls"), err)
+		}
+	if strings.Contains(string(dirPath[0]), "https://github.com/") {
```

Comment:
The Git repo does not have to be `github.com` - could be any Git service, right?

Reply:
validating till https request.